File: 프로그래머스/lv2/오하늘/멀쩡한 사각형.py
# 최대공약수 풀이 참고
import math

def solution(w,h):
    n = math.gcd(w,h) #최대 공약수 구하기
    exp = w+h-n # 대각선이 지나는 정사각형
    total = w*h
    return total - exp # 전체 - 정사각형

File: 프로그래머스/lv2/이하은/멀쩡한 사각형.kt
/*
val cnt = gdc(w, h)
w*h - ((w/cnt + h/cnt - 1) * cnt)
*/
class Solution {
    fun solution(w: Int, h: Int): Long {
        val cnt = gcd(w, h)
        return w*h.toLong() - ((w/cnt + h/cnt - 1) * cnt.toLong())
    }
    
    fun gcd(a: Int, b: Int): Int {
        if(a % b == 0){
            return b
        }else{
            return gcd(b, a % b)
        }
    }
}